modules/custom/az_eds/src/Plugin/KeyProvider/AZPantheonEnvKeyProvider.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drupal\az_eds\Plugin\KeyProvider;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\pantheon_secrets\Plugin\KeyProvider\PantheonSecretKeyProvider;
+use Drupal\key\KeyInterface;
+
+/**
+ * A key provider that fetches a key from Pantheon secrets, or the environment.
+ *
+ * @KeyProvider(
+ *   id = "az_pantheon_env",
+ *   label = @Translation("Quickstart Pantheon Secret or Environment variable"),
+ *   description = @Translation("This key provider retrieves from a Pantheon secret or a environment variable fallback."),
+ *   storage_method = "pantheon",
+ *   key_value = {
+ *     "accepted" = FALSE,
+ *     "required" = FALSE
+ *   }
+ * )
+ */
+class AZPantheonEnvKeyProvider extends PantheonSecretKeyProvider {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+    // If not on pantheon, secret_name should become text input.
+    if (!defined('PANTHEON_ENVIRONMENT')) {
+      $form['secret_name'] = [
+        '#type' => 'textfield',
+        '#title' => $this->t('Secret name or environment variable'),
+        '#description' => $this->t('Name of the secret or environment variable.'),
+        '#required' => TRUE,
+        '#default_value' => $this->getConfiguration()['secret_name'],
+      ];
+    }
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $key_provider_settings = $form_state->getValues();
+    $secret_name = $key_provider_settings['secret_name'];
+    // Don't throw an error for a missing secret.
+    if (defined('PANTHEON_ENVIRONMENT')) {
+      $secret_value = $this->secretsClient->getSecret($secret_name);
+
+      // Does the secret exist.
+      if (!$secret_value) {
+        $form_state->setErrorByName('secret_name', $this->t('The secret does not exist or it is empty.'));
+        return;
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getKeyValue(KeyInterface $key) {
+    // If we're on pantheon, use secrets exclusively.
+    if (defined('PANTHEON_ENVIRONMENT')) {
+      return parent::getKeyValue($key);
+    }
+    // Environment variable fallback.
+    // Unlike EnvKeyProvider, we use the pantheon config key of secret_name.
+    $env_variable = $this->configuration['secret_name'];
+    $key_value = getenv($env_variable);
+
+    if (!$key_value) {
+      return NULL;
+    }
+
+    if (isset($this->configuration['base64_encoded']) && $this->configuration['base64_encoded'] === TRUE) {
+      // Phpstan doesn't like base64_decode.
```

Comment:
Interesting . I've definitely seen the sniffs provided by the old phpcs-securiry-audit package that we're still including (and should probably consider removing) complain about stuff like this but I don't think I've ever seen PHPStan complain about it... 🤔

Reply:
It is actually a phpcs complaint. Not sure why I thought it was phpstan...I've removed the phpstan squelch. The specific warning phpcs gives is `WARNING | Crypto function base64_decode used.`